old-tutorials-in-development/tutorials-in-development/R/dc-spatio-temporal-intro/07-spatial-resolution-extent.Rmd

```diff
@@ -65,95 +95,144 @@ directory to the location of the downloaded and unzipped data subsets.
 size or spatial resolution, tells the program how to place or render each pixel
 in 2 dimensional space.  Tools like `R`, using supporting packages such as
 `rgdal`, and associated raster tools have functions that allow you to view and
 define the extent of a new raster.
 
-``` {r view-extent}
-# View the extent of the raster
-DEM@extent
+``` {r raster-extent}
+# load raster
+DSM_HARV <- raster(paste0(wd,"NEON-DS-Airborne-Remote-Sensing/HARV/DSM/HARV_dsmCrop.tif"))
 
-```
-
-<figure>
-    <a href="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/pixelDetail.png">
-    <img src="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/pixelDetail.png"></a>
-    <figcaption>The spatial resolution of a raster refers the size of each cell
-    in meters. This size in turn relates to the area on the ground that the pixel
-    represents.</figcaption>
-</figure>
+# view crs to determine units
+crs(DSM_HARV)
 
+# view the extent of the raster
+extent(DSM_HARV)
+```
 
 <figure>
-    <img src="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/raster_streched_extent.png">
+    <a href="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/raster_streched_extent.png">
+    <img src="https://raw.githubusercontent.com/NEONScience/NEON-Data-Skills/dev-aten/graphics/raster-general/raster_streched_extent.png"
+    alt="The image on the left has an extent of (8,8) and is less blurry than the image on the right which is stretched to an extent of (16,16)."></a>
     <figcaption>If you double the extent value of a raster - the pixels will be
     stretched over the larger area making it look more "blury".
     </figcaption>
 </figure>
 
 ### Calculating Raster Extent
-To calculate the extent of a raster, we first need the bottom left x,y
-coordinate of the raster. In
-the case of the UTM coordinate system which is in meters, to calculate
-the raster's extent, we can add the number of columns and rows to the x,y corner
-coordinate location of the raster, multiplied by the resolution (the pixel size)
-of the raster.
+To calculate the extent of a raster, we first need the bottom left "X,Y"
+coordinate of the raster. In the case of the UTM coordinate system which is in 
+meters, to calculate the raster's extent, we can add the number of columns and 
+rows to the "X,Y" corner coordinate location of the raster, multiplied by the 
+resolution (the pixel size) of the raster.
 
-Let's explore that next.
+Let's explore that next, using a blank raster that we create.
 
-``` {r calculate-raster-extent}
-# create a raster from the matrix
+``` {r calculate-raster-extent, fig.cap="16 raster cells in various colors plotted between -180 and 180 on the x-axis and -90 and 90 on the y-axis"}
+# create a "blank" raster from the matrix
 myRaster1 <- raster(nrow=4, ncol=4)
 
 # assign some random data to the raster
```

Comment:
These data are not really 'random', but a list of integers from 1:ncell(myRaster1) 

You could populate with 'random' using rnorm() but I like the 1:16 better (because you can keep the image oriented and it also shows how the pixel values are assigned by row/column)